### ISSUE TITLE
git-master: ZynAddSubFx own interface does not control anything

### ISSUE THREAD

**Issue by Fak3** [falkTX/Carla#268]
I installed Carla from current master branch (3ff1404572204f2c5fe8930ad6169fd3ff469dcf)
When i add ZynAddSubFx to rack, it can produce simple sine sound, but if i open its user interface to change instrument - i can't change anything. This ui does not show output signal levels in the bottom bar, when it plays sound. Volume slider is set to zero, but does not change anything. All other controls and buttons do nothing, it is impossible to add banks directory - the list is always empty after i select the dir.
All UI of ZynAddSubFx look sluggish and unresponsive.
Sometimes it takes very long time to show the UI window (about 30 seconds)

I tested some previous version of Carla (2.0beta4 as it is named in openSUSE repo), and it worked fine.

**Comment by falkTX** [comment]
You can keep using the old version if this is important for you.
The latest zyn is a WIP which doesn't have everything working yet.

**Comment by Fak3** [comment]
This bug is still present in current git-master (200a9884b94f44fa42ea371a6067d22670e0d37d)

Console log:
```
Trying to start UI using "/usr/lib/lv2/carla.lv2//resources//zynaddsubfx-ui"
ALL OK!
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
Trying to get window...
```
Zyn UI takes almost a minute to show up, and as described above - lives its own life without affecting generated sound.

After i close the Zyn UI console error appears:

    Carla assertion failure: "pData->pipeRecv != INVALID_PIPE_VALUE" in file ../../utils/CarlaPipeUtils.cpp, line 1012

**Comment by falkTX** [comment]
Which system are you using?

please install the latest zynaddsubfx (git version) and see if works standalone.

**Comment by Fak3** [comment]
I am using openSUSE 13.2.

I installed latest standalone zunaddsubfx from git, and it works fine

**Comment by falkTX** [comment]
This might be something related to liblo and localhost. (I've seen it happening before)

Are you able to show DSSI plugin UIs in Carla?

**Comment by Fak3** [comment]
I have few DSSI plugins installed, and able to run them with jack-dssi-host. Plugins work (i can hear sound from xsynth, for ex.), but i can't see the UI this way - it never shows up, despite the console messages telling it is starting:
```
u1@pc1:~> jack-dssi-host fluidsynth-dssi.so

jack-dssi-host: OSC URL is:
osc.udp://pc1.ll:15920/dssi/fluidsynth-dssi/FluidSynth-DSSI/chan00

host: Ready
fsd-gui starting (pid 9901)...

(FluidSynth-DSSI_gtk:9901): Pango-WARNING **: /usr/lib/pango/1.8.0/modules/pango-basic-fc.so: wrong ELF class: ELFCLASS32


^Cjack-dssi-host: signal 2 caught, trying to clean up and exit
```
 I tried fluidsynth-dssi.so xsynth-dssi.so and hexter.so

In Carla, i can't even see DSSI plugins in the list, except those, which came from dssi-vst bridge:

```
RemoteVSTClient: executing /usr/local/lib/dssi/dssi-vst/dssi-vst-scanner.exe /tmp/rplugin_qry_jvZbgT
fixme:ntoskrnl:ExInitializeResourceLite stub: 0x11113d
fixme:ntoskrnl:KeInitializeSpinLock stub: 0x111195
fixme:ntoskrnl:KeInitializeSpinLock stub: 0x1111a1
fixme:ntoskrnl:IoGetDeviceObjectPointer stub: L"\\Device\\ScsiPort0" 80 0x53e670 0x53e684
fixme:ntoskrnl:KeInitializeSpinLock stub: 0x111189
fixme:ntoskrnl:ExInitializeZone stub: 0x111179, 176, 0x112a18, 22536
fixme:ntoskrnl:IoGetDeviceObjectPointer stub: L"\\Device\\ScsiPort0" 80 0x53e67c 0x53e648
dssi-vst-scanner: found in /home/u1/.vst/RoVee.dll
dssi-vst-scanner: found in /home/u1/.vst/TAL-Vocoder-2.dll
```
I only see RoVee and TAL-Vocoder in DSSI list

**Comment by falkTX** [comment]
Ok, then I guess I know what the issue is - liblo doesn't work in your system.
(All DSSI UIs that I know use liblo)

If jack-dssi-host doesn't work, the zyn plugin in carla will not work either.

Here's some related discussion on the liblo mailing list:
https://sourceforge.net/p/liblo/mailman/message/27326811/

I have no solution at this point.
Perhaps I should include liblo in carla's code to ensure it uses ipv4...

**Comment by amsynthdotcom** [comment]
Don't know if it helps, but for the .so apps like Xsynth, I use Ghostess. Works great!


brian

**Comment by Fak3** [comment]
Yes! Thanks for the info! I finally managed to fix the subject issue - zynaddsubfx UI now works in Carla.

The issue was that liblo generated URL was unresolvable:

    osc.udp://pc1.ll:14345/dssi/xsynth-dssi/Xsynth/chan00

I cahnged my home network settings some time ago, but somehow liblo is still using my old hostname in URL (pc1.ll), which is not valid anymore.

For now i just added this hostname to my /etc/hosts:

    127.0.0.1       pc1.ll

Now all the DSSI plugins UI work properly in jack-dssi-host and zyn UI works in carla. Although i still can't see DSSI plugins in carla list, but it is another story.
